frontend/pages/shelter/pet/add.tsx

```diff
@@ -0,0 +1,231 @@
+import { EditOutlined } from "@ant-design/icons";
+import { Breadcrumb, Button, RadioChangeEvent } from "antd";
+import { Content } from "antd/lib/layout/layout";
+import Title from "antd/lib/typography/Title";
+import { FurLength, PetData, Sex, Species, Status, Sterilised } from "common/enums";
+import {
+	FormSection,
+	ImageSection
+} from "components/shelter/pet/add/FormComponents";
+import ShelterLayout from "layouts/shelter/ShelterLayout";
+import moment from "moment";
+import React, { ChangeEvent, useCallback, useMemo, useState } from "react";
+import styled from "styled-components";
+
+export default function AddNewPet() {
+	const [pet, setPet] = useState<PetData>({
+		key: "",
+		name: "",
+		images: [],
+		visible: true,
+		sex: Sex.MALE,
+		species: Species.CAT,
+		status: Status.HEALTHY,
+		acquired: new Date(),
+		breed: "",
+		furLength: FurLength.SHORT,
+		medicalIssues: [],
+		sterilised: Sterilised.YES,
+		dateOfBirth: new Date(),
+		furColor: [],
+		toiletTrained: true
+	});
+
+	const onValueChange = (e: ChangeEvent<HTMLInputElement>, key: keyof Pick<
+		PetData,
+		"name"
+	>) => {
+		setPet((prev) => ({ ...prev, [key]: e.target.value }));
+	}
+
+	const onRadioChange = (
+		e: RadioChangeEvent,
+		key: keyof Pick<
+			PetData,
+			"visible" | "toiletTrained" | "sex" | "sterilised" | "status"
+		>,
+		isYesNo?: boolean
+	) => {
+		if (!e.target.value) return;
+		const val = isYesNo
+			? e.target.value === "yes"
+				? true
+				: false
+			: e.target.value;
+		setPet((prev) => ({ ...prev, [key]: val }));
+	};
+
+	const onSelectChange = (
+		value: string | string[],
+		key: keyof Pick<
+			PetData,
+			"species" | "furLength" | "breed" | "medicalIssues" | "furColor"
+		>
+	) => {
+		if (!value) return;
+		setPet((prev) => ({ ...prev, [key]: value }));
+	};
+
+	const onDateChange = (
+		date: moment.Moment,
+		key: keyof Pick<PetData, "acquired" | "dateOfBirth">
+	) => {
+		if (!date) return;
+		setPet((prev) => ({ ...prev, [key]: date.toDate() }));
+	};
+
+	const updateGallery = useCallback((images: string[]) => {
```

Comment:
what is the purpose of useCallback here?

Reply:
To stop the recursive update. I'm not 100% sure on how to solve this actually - not sure if my implementation is correct. @ryhuz please help me take a look again leh 😥